src/Components/Components/src/Routing/RouteContext.cs

```diff
@@ -15,6 +16,11 @@ public RouteContext(string path)
             // This is a simplification. We are assuming there are no paths like /a//b/. A proper routing
             // implementation would be more sophisticated.
             Segments = path.Trim('/').Split(Separator, StringSplitOptions.RemoveEmptyEntries);
+            // Individual segments are URL-decoded in order to support arbitrary characters, assuming UTF-8 encoding.
+            for (int i = 0; i < Segments.Length; i++)
+            {
+                Segments[i] = WebUtility.UrlDecode(Segments[i]);
```

Comment:
@Tratcher This is for client side blazor. (Assume no ASP.NET Core)
Is this the best way to decode the segments? (vs `Uri.UnescapeDataString`.  or any other package with minimal dependencies we can bring in).

Reply:
I believe we should use https://www.nuget.org/packages/System.Text.Encodings.Web.

@SteveSandersonMS This will bring an additional package, but I think its important to use this one as it is the blessed one and other methods have nuances that we would need to understand (and would potentially introduce incompatibilities between server and client code)

Reply:
@javiercn How much does it increase the default compressed post-linked app size?

Reply:
@SteveSandersonMS I would have to check if we decide to go this route, but the dll is around 54K uncompressed, so I think it'll be fine.

Independent of size, I think it's worth using it as it ensure compatibility with the server and is the blessed implementation.

Reply:
Where is `path` coming from? What guarantees do we have about it?

Reply:
It’s the url from the browser as returned by window.location.href

Reply:
@javiercn I totally agree that it's ideal to share this logic from the `System.Text.Encodings.Web` package. From what you're saying about the 54K uncompressed unlinked size, it sounds like we shouldn't have any problem with that. Nuget.org says it doesn't have any further dependencies.

Reply:
@SteveSandersonMS Let's switch to that then. We probably will see more gains when we link it.

Reply:
@javiercn Thanks, I wasn't aware of this package. I thought `WebUtility` was still the correct replacement for `HttpUtility`. 😅
I'll switch to this implementation and see how it works.

Reply:
It took me a bit of time to upgrade my setup and rebuild the whole solution… And now that I've looked at the contents of the package, it seems that it only provides methods for encoding. 😅

See the source here for reference : https://github.com/dotnet/corefx/tree/9e2f3b630a51875fc0455a7cabb8803f6527f81e/src/System.Text.Encodings.Web/src/System/Text/Encodings/Web

In that case, we should pick either `WebUtility.UrlDecode` or `Uri.UnescapeDataString`.
The main difference between `Uri.UnescapeDataString` and `WebUtility.UrlDecode`, would be that the former doesn't convert `+` into space (cf. [documentation](https://docs.microsoft.com/en-us/dotnet/api/system.uri.unescapedatastring)), while the latter does.

In this case, `Uri.UnescapeDataString` might indeed have been the better choice. A quick search in the repository reveals that it is used in quite a few places already: https://github.com/aspnet/AspNetCore/search?q=UnescapeDataString&unscoped_q=UnescapeDataString

If that's ok with you, I'll make that change instead.

Reply:
UnescapeDataString should be fine for individual path segments.